commit 7e00441c257e7e9e5dc5ab983fc06046fb72b0c5
Author: Gilles Caulier <caulier.gilles@gmail.com>
Date:   Sat Jul 22 15:46:08 2017 +0200

    fix broken linking stage under MacOS with macports.
    move database models into libdigikamdatabase.
    Let's others model in place to be included into libdigikamcore

diff --git a/libs/database/CMakeLists.txt b/libs/database/CMakeLists.txt
index 7d055363fb..a431a36d7b 100644
--- a/libs/database/CMakeLists.txt
+++ b/libs/database/CMakeLists.txt
@@ -13,6 +13,18 @@ endif (POLICY CMP0063)
 # Boost uses operator names (and, not, ...)
 string(REPLACE "-fno-operator-names" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
 
+set(libdatabasemodels_SRCS
+    models/imagemodel.cpp
+    models/imagefiltermodel.cpp
+    models/imagefiltermodelpriv.cpp
+    models/imagefiltermodelthreads.cpp
+    models/imagefiltersettings.cpp
+    models/imagelistmodel.cpp
+    models/imagesortsettings.cpp
+    models/imagethumbnailmodel.cpp
+    models/imageversionsmodel.cpp
+)
+
 set(libdatabasecore_SRCS
     server/databaseserverstarter.cpp
     server/databaseservererror.cpp
@@ -152,10 +164,10 @@ if(ENABLE_DBUS)
     include_directories($<TARGET_PROPERTY:Qt5::DBus,INTERFACE_INCLUDE_DIRECTORIES>)
 endif()
 
-add_library(digikamdatabase_src     OBJECT ${digikamdatabase_LIB_SRCS})
+add_library(digikamdatabase_src     OBJECT ${digikamdatabase_LIB_SRCS} ${libdatabasemodels_SRCS})
 add_library(digikamdatabasemain_src OBJECT ${libdatabaseutils_SRCS} ${libimgqsort_SRCS})
 add_library(digikamdatabasecore_src OBJECT ${libdatabasecore_SRCS})
-add_library(digikamdatabase         SHARED $<TARGET_OBJECTS:digikamdatabase_src> $<TARGET_OBJECTS:digikamdatabasemodels_src>)
+add_library(digikamdatabase         $<TARGET_OBJECTS:digikamdatabase_src>)
 
 generate_export_header(digikamdatabase
                        BASE_NAME digikam_database
diff --git a/libs/models/imagefiltermodel.cpp b/libs/database/models/imagefiltermodel.cpp
similarity index 100%
rename from libs/models/imagefiltermodel.cpp
rename to libs/database/models/imagefiltermodel.cpp
diff --git a/libs/models/imagefiltermodel.h b/libs/database/models/imagefiltermodel.h
similarity index 100%
rename from libs/models/imagefiltermodel.h
rename to libs/database/models/imagefiltermodel.h
diff --git a/libs/models/imagefiltermodelpriv.cpp b/libs/database/models/imagefiltermodelpriv.cpp
similarity index 100%
rename from libs/models/imagefiltermodelpriv.cpp
rename to libs/database/models/imagefiltermodelpriv.cpp
diff --git a/libs/models/imagefiltermodelpriv.h b/libs/database/models/imagefiltermodelpriv.h
similarity index 100%
rename from libs/models/imagefiltermodelpriv.h
rename to libs/database/models/imagefiltermodelpriv.h
diff --git a/libs/models/imagefiltermodelthreads.cpp b/libs/database/models/imagefiltermodelthreads.cpp
similarity index 100%
rename from libs/models/imagefiltermodelthreads.cpp
rename to libs/database/models/imagefiltermodelthreads.cpp
diff --git a/libs/models/imagefiltermodelthreads.h b/libs/database/models/imagefiltermodelthreads.h
similarity index 100%
rename from libs/models/imagefiltermodelthreads.h
rename to libs/database/models/imagefiltermodelthreads.h
diff --git a/libs/models/imagefiltersettings.cpp b/libs/database/models/imagefiltersettings.cpp
similarity index 100%
rename from libs/models/imagefiltersettings.cpp
rename to libs/database/models/imagefiltersettings.cpp
diff --git a/libs/models/imagefiltersettings.h b/libs/database/models/imagefiltersettings.h
similarity index 100%
rename from libs/models/imagefiltersettings.h
rename to libs/database/models/imagefiltersettings.h
diff --git a/libs/models/imagelistmodel.cpp b/libs/database/models/imagelistmodel.cpp
similarity index 100%
rename from libs/models/imagelistmodel.cpp
rename to libs/database/models/imagelistmodel.cpp
diff --git a/libs/models/imagelistmodel.h b/libs/database/models/imagelistmodel.h
similarity index 100%
rename from libs/models/imagelistmodel.h
rename to libs/database/models/imagelistmodel.h
diff --git a/libs/models/imagemodel.cpp b/libs/database/models/imagemodel.cpp
similarity index 100%
rename from libs/models/imagemodel.cpp
rename to libs/database/models/imagemodel.cpp
diff --git a/libs/models/imagemodel.h b/libs/database/models/imagemodel.h
similarity index 100%
rename from libs/models/imagemodel.h
rename to libs/database/models/imagemodel.h
diff --git a/libs/models/imagesortsettings.cpp b/libs/database/models/imagesortsettings.cpp
similarity index 100%
rename from libs/models/imagesortsettings.cpp
rename to libs/database/models/imagesortsettings.cpp
diff --git a/libs/models/imagesortsettings.h b/libs/database/models/imagesortsettings.h
similarity index 100%
rename from libs/models/imagesortsettings.h
rename to libs/database/models/imagesortsettings.h
diff --git a/libs/models/imagethumbnailmodel.cpp b/libs/database/models/imagethumbnailmodel.cpp
similarity index 100%
rename from libs/models/imagethumbnailmodel.cpp
rename to libs/database/models/imagethumbnailmodel.cpp
diff --git a/libs/models/imagethumbnailmodel.h b/libs/database/models/imagethumbnailmodel.h
similarity index 100%
rename from libs/models/imagethumbnailmodel.h
rename to libs/database/models/imagethumbnailmodel.h
diff --git a/libs/models/imageversionsmodel.cpp b/libs/database/models/imageversionsmodel.cpp
similarity index 100%
rename from libs/models/imageversionsmodel.cpp
rename to libs/database/models/imageversionsmodel.cpp
diff --git a/libs/models/imageversionsmodel.h b/libs/database/models/imageversionsmodel.h
similarity index 100%
rename from libs/models/imageversionsmodel.h
rename to libs/database/models/imageversionsmodel.h
diff --git a/libs/models/CMakeLists.txt b/libs/models/CMakeLists.txt
index cbabfaa9b8..804456b955 100644
--- a/libs/models/CMakeLists.txt
+++ b/libs/models/CMakeLists.txt
@@ -9,18 +9,6 @@ if (POLICY CMP0063)
     cmake_policy(SET CMP0063 NEW)
 endif (POLICY CMP0063)
 
-set(libdatabasemodels_SRCS
-    imagemodel.cpp
-    imagefiltermodel.cpp
-    imagefiltermodelpriv.cpp
-    imagefiltermodelthreads.cpp
-    imagefiltersettings.cpp
-    imagelistmodel.cpp
-    imagesortsettings.cpp
-    imagethumbnailmodel.cpp
-    imageversionsmodel.cpp
-)
-
 set(libalbummodels_SRCS
     imagealbummodel.cpp
     imagealbumfiltermodel.cpp
@@ -52,5 +40,4 @@ endif()
 #for digikam core lib
 add_library(digikamgenericmodels_src OBJECT ${libgenericmodels_SRCS})
 
-add_library(digikamdatabasemodels_src OBJECT ${libdatabasemodels_SRCS})
-add_library(digikammodels_src OBJECT  ${libalbummodels_SRCS} ${libgenericmodels_SRCS})
+add_library(digikammodels_src OBJECT ${libalbummodels_SRCS} ${libgenericmodels_SRCS})

From c51043283aa9d2612b88afe0146f0d63c77b08ed Mon Sep 17 00:00:00 2001
From: Simon Frei <freisim93@gmail.com>
Date: Wed, 2 Aug 2017 12:12:55 +0200
Subject: cmake: Fix warning: Remove ECMPackageConfigHelpers (unused)

---
 CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2aaf56b..9a48492 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -80,7 +80,6 @@ include(ECMAddTests)
 include(ECMMarkNonGuiExecutable)
 include(ECMGenerateHeaders)
 include(ECMGeneratePriFile)
-include(ECMPackageConfigHelpers)
 include(ECMSetupVersion)
 include(ECMInstallIcons)
 include(ECMAddAppIcon)
-- 
cgit v0.11.2
